src/Form/Form.js

```diff
@@ -297,8 +301,7 @@ class Form {
             let canExecute = true
             // sometimes we want to execute even if we didn't decorate
             if (shouldCheckForDecorate) {
-                const {shouldDecorate} = getInputConfig(input)
-                canExecute = shouldDecorate(input, this)
+                canExecute = isFieldDecorated(input)
```

Comment:
Instead of checking if the field should be decorated (which is now an async operation) we check if the field is already decorated, which is what we actually wanted.